### PR TITLE
✨ Bento `amp-gist`

### LINKED ISSUE #16: Bento amp-gist
Create a Bento component for [amp-gist](https://amp.dev/documentation/examples/components/amp-gist/)

### PR DESCRIPTION
**Pull request by deepaklalwani97**
This PR adds a bento component for `amp-gist.

- [x] Add `amp-gist`
- [x] Add types
- [x] Add type annotations
- [x]  Add test case
- [x]  Update Storybook
- [x]  prettify new `amp-gist`

### CODE REVIEW

**Review by AnuragVasanwala — APPROVED** [review]
LGTM.

File: extensions/amp-gist/1.0/test/validator-amp-gist.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <style amp-custom>
+      amp-gist {
+        color: red;
+      }
+    </style>
+    <script async custom-element="amp-gist" src="https://cdn.ampproject.org/v0/amp-gist-1.0.js"></script>
+    <script async src="https://cdn.ampproject.org/v0.js"></script>
+  </head>
+  <body>
+    <amp-gist layout="responsive" width="150" height="80"></amp-gist>
```
Comment (ediamin) [inline extensions/amp-gist/1.0/test/validator-amp-gist.html]: No `data-gistid` or `data-file` attribute. Can we duplicate the`validator-amp-gist.html` from v0.1 and use it here?

File: extensions/amp-gist/1.0/test/test-amp-gist.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      element = createElementWithAttributes(win.document, 'amp-gist', {
+        'data-gistid': 'b9bb35bc68df68259af94430f012425f',
+        'height': 500,
+        'layout': 'fixed-height',
+      });
+      doc.body.appendChild(element);
+      await waitForRender();
+
+      expect(element.shadowRoot.querySelector('iframe').src).not.to.be.null;
+    });
```
Comment (ediamin) [inline extensions/amp-gist/1.0/test/test-amp-gist.js:48]: Please add a test case for `data-file`.

File: extensions/amp-gist/1.0/storybook/Basic.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import {BentoGist} from '../component';
+const GIST_ID = 'b9bb35bc68df68259af94430f012425f';
+export default {
+  title: 'Gist',
+  component: BentoGist,
+};
+
+export const _default = () => {
+  return <BentoGist style={{height: 500}} gistId={GIST_ID}></BentoGist>;
+};
```
Comment (ediamin) [inline extensions/amp-gist/1.0/storybook/Basic.js:26]: Please add an example for `file` prop.

File: extensions/amp-gist/1.0/storybook/Basic.amp.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+export const _default = () => {
+  return (
+    <amp-gist
+      data-gistid="b9bb35bc68df68259af94430f012425f"
+      layout="fixed-height"
+      height="300"
+    ></amp-gist>
+  );
```
Comment (ediamin) [inline extensions/amp-gist/1.0/storybook/Basic.amp.js:21]: Please add an example for data-file attribute.

File: extensions/amp-gist/1.0/component.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import {MessageType_Enum, deserializeMessage} from '#core/3p-frame-messaging';
+
+import * as Preact from '#preact';
+import {useCallback, useEffect, useMemo, useRef, useState} from '#preact';
+import {ProxyIframeEmbed} from '#preact/component/3p-frame';
+
+const TYPE = 'github';
+const DEFAULT_TITLE = 'Github gist';
```
Comment (ediamin) [inline extensions/amp-gist/1.0/component.js]:
```suggestion
const DEFAULT_TITLE = 'GitHub Gist';
```

**Review by ediamin — CHANGES_REQUESTED** [review]

**Review by ediamin — APPROVED** [review]
LGTM. Great work! 👍

**Comment by AnuragVasanwala** [comment]
This component is merged 🎉

**Pull request closed without merging**